dataset/dataset.cpp

```diff
@@ -573,104 +475,216 @@ void DataArrayView::setUnit(const units::Unit unit) const {
-AttrsConstView DataArrayConstView::attrs() const noexcept {
-  return makeView<AttrsConstView>();
+/// Return a view to all coordinates of the data view.
+CoordsView DataArrayView::coords() const noexcept {
+  // Typically view of item of data array, therefore:
+  // ds['a'].coords['x'] = x # inserts unaligned coord
+  return make_coords(*this, CoordCategory::All);
 }
 
+/// Return a const view to all coordinates of the data array.
+CoordsConstView DataArray::coords() const { return get().coords(); }
+
+/// Return a view to all coordinates of the data array.
+CoordsView DataArray::coords() {
+  // Note difference to DataArrayView::coords():
+  // array.coords['x'] = x # inserts aligned coord
+  return make_coords(get(), CoordCategory::All, false);
+}
+
+/// Return a const view to all aligned coordinates of the data view.
+CoordsConstView DataArrayConstView::aligned_coords() const noexcept {
+  return make_coords(*this, CoordCategory::Aligned);
+}
+
+/// Return a view to all aligned coordinates of the data view.
+CoordsView DataArrayView::aligned_coords() const noexcept {
+  return make_coords(*this, CoordCategory::Aligned);
+}
+
+/// Return a const view to all aligned coordinates of the data array.
+CoordsConstView DataArray::aligned_coords() const {
+  return get().aligned_coords();
+}
+
+/// Return a view to all aligned coordinates of the data array.
+CoordsView DataArray::aligned_coords() { return m_holder.coords(); }
+
+/// Return a const view to all unaligned coordinates of the data view.
+CoordsConstView DataArrayConstView::unaligned_coords() const noexcept {
+  return make_coords(*this, CoordCategory::Unaligned);
+}
+//
+/// Return a view to all unaligned coordinates of the data view.
+CoordsView DataArrayView::unaligned_coords() const noexcept {
+  return make_coords(*this, CoordCategory::Unaligned);
+}
+
+/// Return a const view to all unaligned coordinates of the data array.
+CoordsConstView DataArray::unaligned_coords() const {
+  return get().unaligned_coords();
+}
+
+/// Return a view to all unaligned coordinates of the data array.
+CoordsView DataArray::unaligned_coords() { return get().unaligned_coords(); }
+
 /// Return a const view to all masks of the data view.
 MasksConstView DataArrayConstView::masks() const noexcept {
-  return makeView<MasksConstView>();
+  const bool is_view_of_unaligned = !get_data().data && hasData();
+  auto items = is_view_of_unaligned
+                   ? makeViewItems(parentDims(), get_data().masks)
+                   : makeViewItems(get_data().masks);
+  if (is_view_of_unaligned) {
+    // This is a view of the unaligned content of a realigned data array.
+    decltype(*this) unaligned = get_data().unaligned->data;
+    const auto m = unaligned.masks();
+    items.insert(m.items().begin(), m.items().end());
+  }
+  return MasksConstView(std::move(items), slices());
+}
+
+/// Return a view to all masks of the data view.
+MasksView DataArrayView::masks() const noexcept {
```

Comment:
Should be possible, but adds some extra if statements and does not really aid in readability. Since I am just considering changes to how `realign` is handled (which might mean that 2/3 of this function is not necessary) I suggest to leave it for now.